File: src/server/SmartSolarMicrogrid.Api/DTOs/DashboardMetricsDto.cs
// Description: Data Transfer Objects representing operational dashboard metrics and active booking spotlight data.

namespace SmartSolarMicrogrid.Api.DTOs;

/// <summary>
/// Description: Aggregated response payload for GET /api/reservations/dashboard-metrics.
/// </summary>
public class DashboardMetricsResponseDto
{
    public int PendingReservationsCount { get; set; }

    public int ApprovedFutureReservationsCount { get; set; }

    public int CompletedTodayCount { get; set; }

    public ActiveSpotlightDto? ActiveSpotlight { get; set; }
}

/// <summary>
/// Description: Embedded DTO representing the nearest upcoming approved reservation spotlight.
/// </summary>
public class ActiveSpotlightDto
{
    public string ReservationId { get; set; } = string.Empty;

    public string StationName { get; set; } = string.Empty;

    public string AllocatedBayId { get; set; } = string.Empty;

    public DateTime ScheduledDateTime { get; set; }

    public string Status { get; set; } = string.Empty;

    public double EstimatedKwh { get; set; }
}

File: src/mobile-android/SmartSolarMicrogridAndroid/app/src/main/kotlin/com/sliit/ssmts/operator_dashboard/data/remote/dto/DashboardMetricsDto.kt
/**
 * Description: Data Transfer Objects representing operational dashboard metrics and active booking spotlight responses.
 */
package com.sliit.ssmts.operator_dashboard.data.remote.dto

import com.google.gson.annotations.SerializedName

/**
 * Description: Encapsulates live count metrics and spotlight booking returned by GET /api/reservations/dashboard-metrics.
 */
data class DashboardMetricsDto(
    @SerializedName("pendingReservationsCount")
    val pendingReservationsCount: Int,

    @SerializedName("approvedFutureReservationsCount")
    val approvedFutureReservationsCount: Int,

    @SerializedName("completedTodayCount")
    val completedTodayCount: Int,

    @SerializedName("activeSpotlight")
    val activeSpotlight: ActiveSpotlightDto? = null
)

/**
 * Description: Encapsulates the nearest upcoming approved reservation details for the active spotlight card.
 */
data class ActiveSpotlightDto(
    @SerializedName("reservationId")
    val reservationId: String,

    @SerializedName("stationName")
    val stationName: String,

    @SerializedName("allocatedBayId")
    val allocatedBayId: String,

    @SerializedName("scheduledDateTime")
    val scheduledDateTime: String,

    @SerializedName("status")
    val status: String,

    @SerializedName("estimatedKwh")
    val estimatedKwh: Double
)
